### ISSUE TITLE
How can I configure storybook as server side on react.

### ISSUE THREAD

**Issue by ankitstudent09** [storybooks/storybook#4271]
I am using server side react and therein I am unable to use my storybook in my existing project, so how can I use storybook as server side react and what all changes need to make.

**Comment by ankitstudent09** [comment]
Below is my sample app.js file from where my react application is started on server 3000

**app.js**
```

if (process.env.NODE_ENV === 'development') {

    const webpackConfig = require('./webpack.config'),
          compiler = webpack(webpackConfig);

    app.listen(3000, '0.0.0.0', function(req, res) {
        // print a message when the server starts listening
        console.log("server starting on 3000");
        //process.env.ORIGIN = req.get('origin');
    });
    app.use(webpackDevMiddleware(compiler, {
        publicPath: webpackConfig.output.publicPath,
        stats: {colors: true}
    }));

    app.use(webpackHotMiddleware(compiler, {
        log: console.log
    }))

}

```

**Comment by igor-dv** [comment]
I am not sure I got your question. Do you want to build a static version of Storybook and serve it with your app?

**Comment by ankitstudent09** [comment]
I am using node with react, from node application I am connecting my graphQL, redis and mongodb. Components created in react are communicating via node for fetching up the output for my component, when I am using storybook as per described to start are as follows:-
```javascript
npm start storybook
```
and in **stories/index.js**, where I am binding my component with storybook doesn't fetch the Initial content from node application as it's not started and configured with _npm start storybook_

Hence I need a work around wherein I can start storybook on same port inside my node application so as to consume my node services in react component.

 below I am attaching the **app.js** code base

```javascript
import express from 'express';
import React from 'react';
import { renderToString } from 'react-dom/server';
import App from './client/App';
import Html from './client/Html';

const port = 3000;
const server = express();

server.get('/', (req, res) => {
  /**
   * renderToString() will take our React app and turn it into a string
   * to be inserted into our Html template function.
   */
  const body = renderToString(<App />);
  const title = 'Server side Rendering with Styled Components';

  res.send(
    Html({
      body,
      title
    })
  );
});

server.listen(port);
console.log(`Serving at http://localhost:${port}`);
```

**Comment by igor-dv** [comment]
Can you run your node application and storybook in two different processes?

**Comment by ankitstudent09** [comment]
> Your react components are directly fetching the data from your node application, am I right?

Yes, My React Components fetching data from node and configuring storybook doesn't allow to fetch data from node server and I want to configure storybook in such a way where In I can fetch data from node application.

**Comment by ankitstudent09** [comment]
> Can you run your node application and storybook in two different processes?

Yes we are running storybook and node application in different process but importing the react component in storybook doesn't communicate to node service as they now become difference instance.

**Comment by igor-dv** [comment]
Can you please share an example of such a component?

Anyway, I think it's not possible to integrate the Storybook **dev** (server side) mode to a different node server. You can serve a **static** version though (also, a static version can be run with the watch mode). Or, to make them communicate (I don get it, why can't they do this ?)

Side note:
IMO the most common usage of Storybook is to isolate your components from the application, hence if these components need a state from the outside, you probably want to mock it.

**Comment by ankitstudent09** [comment]
> Can you please share an example of such a component?
> 
> Anyway, I think it's not possible to integrate the Storybook **dev** (server side) mode to a different node server. You can serve a **static** version though (also, a static version can be run with the watch mode). Or, to make them communicate (I don get it, why can't they do this ?)
> 
> Side note:
> IMO the most common usage of Storybook is to isolate your components from the application, hence if these components need a state from the outside, you probably want to mock it.

Thanks for your quick update, I too thinking of same process that I need to isolate my component from Node and pass on my props data for the behavior response. 

Just in my case there been around 10 components that are child component of parent and all are inter-related to node service for configuration and data fetching for the first time.

That includes major rework for that single parent component to make it isolate from Node.

**Comment by igor-dv** [comment]
If you have a fetching logic as a separate module, you can just mock it with babel or webpack (in storybook I mean)
